typescript/examples/model-context-protocol-smart-wallet-server/index.ts

```diff
@@ -57,13 +57,24 @@ async function initializeServer() {
     );
 
     // Configure CDP Wallet Provider
+    const privateKey = (process.env.PRIVATE_KEY || generatePrivateKey()) as Hex;
+    const signer = privateKeyToAccount(privateKey);
+
     const config = {
       apiKeyName: process.env.CDP_API_KEY_NAME!,
       apiKeyPrivateKey: process.env.CDP_API_KEY_PRIVATE_KEY!,
       networkId: process.env.NETWORK_ID || "base-sepolia",
+      smartWalletAddress: process.env.SMART_WALLET_ADDRESS! as Address,
+      signer,
     };
 
-    const walletProvider = await CdpWalletProvider.configureWithWallet(config);
+    const walletProvider = await SmartWalletProvider.configureWithWallet(config);
+
+    if (!process.env.PRIVATE_KEY || !process.env.SMART_WALLET_ADDRESS) {
+      console.log("Save your private key and smart wallet address to the environment variables");
+      console.log("PRIVATE_KEY=" + privateKey);
```

Comment:
Since we're now generating the private key if none provided, and McpServers do not have the permission to write files to the file system, we log it so the user has a chance to save the key for reuse